api/lib/utils.js

```diff
@@ -115,6 +117,34 @@ function isValidLedgerWord(ledger) {
   return (/^current$|^closed$|^validated$/.test(ledger));
 }
 
+function attachDate(api, baseTransactions, callback) {
+  var groupedTx = _.groupBy(baseTransactions, function(tx) {
+    return tx.ledger_index;
+  });
+
+  async.each(_.keys(groupedTx), function(ledger, next) {
```

Comment:
This is pretty hard to read. I think it would be clearer if it looked like
`async.map(_.keys(groupedTx), _.partial(attachDateToLedgerTransactions, groupedTx), callback)` and `attachDateToLedgerTransactions` was defined above `attachDate`.